index.ts

```diff
@@ -135,5 +142,10 @@ export default {
       type: 'natural',
       order: 'asc',
     }),
+    'recommended-custom': createConfig({
+      type: 'custom',
+      order: 'asc',
+    }),
   },
-} as PluginConfig
+  Alphabet,
```

Comment:
Perhaps this shouldn't be in the default export?

Wouldn't it be better to use a named export?

Reply:
@azat-io I mentioned this limit at the end of the PR description in the `Limits` section:

It would be nice for users to be able to do

`import { Alphabet } from 'eslint-plugin-perfectionist';`

Unfortunately, we currently use a default export in the plugin, and we can't add a named export Alphabetical on top of that.

Changing the default export to a named export would be a breaking change, as users would not be able to do `import perfectionist from 'eslint-plugin-perfectionist'`.

If you manage to find a solution, feel free to suggest it!

Reply:
@hugop95 here is the syntax:

```ts
export class Alphabet { ... }

export default { ... }
```

```ts
class Alphabet { ... }

export default { ... }
export { Alphabet }
```

or

```ts
class Alphabet { ... }
const plugin = {}

export { plugin as default, Alphabet }

Reply:
@OlivierZal

The

```ts
import perfectionist from 'eslint-plugin-perfectionist'
```
will not work the same anymore: this will break ESLint for users:

- Current value of `perfectionist`: 
<img width="350" alt="Screenshot 2024-12-08 at 13 47 05" src="https://github.com/user-attachments/assets/57226150-d280-4d94-a0aa-83bb62f625c5">

- Value after the change:
<img width="730" alt="Screenshot 2024-12-08 at 13 46 25" src="https://github.com/user-attachments/assets/bc2a16a4-d8c1-4bf1-9475-127c40d6b5b7">

Reply:
Mmmmmm, I thought it would be fine, I need to perform some tests.

Reply:
Ok, under node_modules the plugin is in commonjs and module.exports doesn't support named exports. For the moment I have issues to convert into pure esm – which should support named exports besides default export.

Reply:
Ok it's not as simple as I thought. The simplest would probably be a separate module as suggested in https://github.com/azat-io/eslint-plugin-perfectionist/pull/404#issuecomment-2510861165

Reply:
Is the problem related to CJS?

How about we make an export from a separate file and add that to `exports` in `package.json`?

Reply:
@azat-io 

> Is the problem related to CJS?

I'll let @OlivierZal answer this 🙂

> How about we make an export from a separate file and add that to exports in package.json?

Added in a100c46845db9a8440205aa94064c1e2c8aebec3:

Users can now import the `Alphabet` class through `import { Alphabet }  from 'eslint-plugin-perfectionist/alphabet'`.